capstone-app/src/app/modules/history/wiki-results.service.ts

```diff
@@ -46,49 +57,80 @@ export class WikiResultsService {
       );
   }
 
-  fixString(text: string): {history: string, furtherReading?: Map<string, string>} {
+  searchNewLang(query: string, language: string, wordForHistory: string): Observable<WikiServiceResult> {
+    return this.http.get<WikiSearchResult>(
+            `${WikiResultsService.URL_BEGINNING}${language}${WikiResultsService.URL_MIDDLE}${query}${WikiResultsService.URL_END}`
+          ).pipe(map(history => {
+          if (history.parse.text) {
+            const langlinks = new Map<string, Map<string, string>>();
+            for (const link of Object.values(history.parse.langlinks)) {
+                const links = new Map<string, string>();
+                links.set('langName', link.langname);
+                links.set('url', link.url);
+                links.set('searchQuery', link['*']);
+                langlinks.set(link.lang, links);
+            }
+            const historyResult = this.fixString(history.parse.text['*'], wordForHistory, language);
+            const title = history.parse.title;
+            return {title, ...historyResult, langlinks};
+          }
+          else {
+            throw new Error('API did not return a valid response');
+          }
+        }),
+        catchError(() => {
+          console.error('API did not return a valid response.');
+          return of({title: '', history: ''});
+        })
+      );
+  }
+
+
+  fixString(text: string, wordForHistory: string, language: string): HistoryFixString {
```

Comment:
let's remove the console.log calls in this method. (I am indifferent to console.error, however)